server/app/controllers/OrderController.ts

```diff
@@ -0,0 +1,91 @@
+import { Request } from 'express';
+import { inject } from 'inversify';
+import {
+  controller,
+  httpGet,
+  httpDelete,
+  results,
+  httpPost,
+  httpPatch,
+} from 'inversify-express-utils';
+import TYPES from '../constants/types';
+import { BaseController } from './BaseController';
+import { OrderService } from '../services/OrderService';
+import { IOrder } from '../models/OrderModel';
+import { MaterialTypes } from '../entities/Material';
+
+@controller('/orders')
+export class OrderController extends BaseController {
+  constructor(@inject(TYPES.OrderService) private orderService: OrderService) {
+    super();
+  }
+
+  @httpGet('/')
+  public async get(): Promise<results.JsonResult> {
+    try {
+      const orders: IOrder[] = await this.orderService.getOrders();
+      return this.json(orders);
+    } catch (err) {
+      return this.handleError(err);
+    }
+  }
+
+  @httpPost('/')
+  public async post(request: Request): Promise<results.JsonResult> {
+    try {
+      const order: IOrder = await this.orderService.createOrder(request.body);
+      return this.json(order);
+    } catch (err) {
+      return this.handleError(err);
+    }
+  }
+
+  @httpDelete('/')
+  public async delete(request: Request): Promise<results.JsonResult> {
+    try {
+      const order: IOrder | null = await this.orderService.deleteOrder(request.body);
+
+      return this.json(order);
+    } catch (err) {
+      return this.handleError(err);
+    }
+  }
+
+  @httpPatch('/:id')
+  public async update(request: Request): Promise<results.JsonResult> {
+    try {
+      const order: IOrder | null = await this.orderService.updateOrder(
+        request.params.id,
+        request.body
+      );
+      return this.json(order);
+    } catch (err) {
+      return this.handleError(err);
+    }
+  }
+
+  @httpGet('/materialList')
+  public async getMaterialList(): Promise<results.JsonResult> {
+    return this.json(MaterialTypes);
```

Comment:
could it be beneficial to change this to the list of materials with the costs? (balls = busted)